code/api/src/main/scala/com/avast/datadog4s/api/MetricFactory.scala

```diff
@@ -1,15 +1,30 @@
 package com.avast.datadog4s.api
 
+import com.avast.datadog4s.api.MetricFactory.TimerMode.HistogramTimer
+import com.avast.datadog4s.api.MetricFactory.TimerMode
 import com.avast.datadog4s.api.metric.{ Count, Timer, UniqueSet }
 
 trait MetricFactory[F[_]] {
   def histogram: HistogramFactory[F]
   def distribution: DistributionFactory[F]
   def gauge: GaugeFactory[F]
-  def timer(aspect: String, sampleRate: Option[Double] = None): Timer[F]
+  def timer(
+    aspect: String,
+    sampleRate: Option[Double] = None,
+    timerMode: TimerMode = HistogramTimer: TimerMode
```

Comment:
This is quite confusing. It took me a while to parse it:) I would suggest to add a new member to `TimerMode`:

```
object TimerMode{
   val historgram: TimerMode = HistrogramTimer
}
```

That way it's not necessary to explicitly provide type.

Reply:
good idea

Reply:
btw for some reason this was only needed in dotty ... in scala2 it compiles fine without the `: TimerMode`